recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -93,60 +105,51 @@ def requirements(self):
             self.requires("ms-gsl/4.0.0")
 
         if self.options.with_abseil:
-            self.requires("abseil/20220623.0", transitive_headers=True)
+            self.requires("abseil/20230125.3", transitive_headers=True)
 
-        if self.options.with_otlp:
-            self.requires("protobuf/3.21.9")
-            if Version(self.version) <= "1.4.1":
-                self.requires("opentelemetry-proto/0.11.0")
-            else:
-                self.requires("opentelemetry-proto/0.20.0")
-
-            if self.options.get_safe("with_otlp_grpc"):
-                self.requires("grpc/1.50.1")
+        if self.options.with_otlp or self.options.with_otlp_grpc:
+            self.requires("protobuf/3.21.12", transitive_headers=True, transitive_libs=True)
+            if self.options.with_otlp_grpc:
+                self.requires("grpc/1.54.3", transitive_headers=True, transitive_libs=True)
```

Comment:
```suggestion
        if self.options.with_otlp_grpc:
            self.requires("protobuf/3.21.12", transitive_headers=True, transitive_libs=True)
            self.requires("grpc/1.54.3", transitive_headers=True, transitive_libs=True)
```

Please, do not consider a deprecated option as part of recipe conditions.